silq/pulses/pulse_types.py

```diff
@@ -668,6 +668,138 @@ def get_voltage(self, t: Union[float, Sequence]) -> Union[float, np.ndarray]:
         return waveform
 
 
+class MultiSinePulse(Pulse):
+    """MultiSinusoidal pulse
+
+    Parameters:
+        name: Pulse name
+        frequency: main (local oscillator) Pulse frequency
+        frequencies: list of Pulse frequencies
+        phases: list of Pulse phases
+        amplitude: Pulse amplitude. If not set, power must be set.
+        amplitudes: list of Pulse amplitudes
+        power: Pulse power. If not set, amplitude must be set.
+        offset: amplitude offset, zero by default
+        frequency_sideband: Mixer sideband frequency (off by default).
+        sideband_mode: Sideband frequency to apply. This feature must
+            be existent in interface. Not used if not set.
+        phase_reference: What point in the the phase is with respect to.
+            Can be two modes:
+            - 'absolute': phase is with respect to `Pulse.t_start`.
+            - 'relative': phase is with respect to t=0 (phase-coherent).
+
+        **kwargs: Additional parameters of `Pulse`.
+
+    Notes:
+        Either amplitude or power must be set, depending on the instrument
+        that should output the pulse.
+    """
+    def __init__(self,
+                 name: str = None,
+                 frequency: float = None,
+                 frequencies: List[float] = None,
+                 phases: List[float] = None,
+                 amplitude: float = None,
+                 amplitudes: List[float] = None,
+                 power: float = None,
+                 offset: float = None,
+                 frequency_sideband: float = None,
+                 sideband_mode: float = None,
+                 phase_reference: str = None,
+                 **kwargs):
+
+        super().__init__(name=name, **kwargs)
+
+        self.frequency = Parameter(initial_value=frequency, unit='Hz',
+                                   set_cmd=None, vals=vals.Numbers())
+        self.frequencies = Parameter(initial_value=frequencies, unit='Hz',
+                                     set_cmd=None, vals=vals.Lists())
+        self.phases = Parameter(initial_value=phases, unit='deg', set_cmd=None,
+                                vals=vals.Lists())
+        self.power = Parameter(initial_value=power, unit='dBm', set_cmd=None,
+                               vals=vals.Numbers())
+        self.amplitude = Parameter(initial_value=amplitude, unit='V',
```

Comment:
Why does it also have a single amplitude?

Reply:
I left it to be able to define microwave power from this amplitude, but I don't think it is implemented in the interface plus we never use this, so I agree, I will remove this

Reply:
Corrected.